code/game/machinery/recharger.dm

```diff
@@ -110,13 +110,10 @@
 		if (istype(G, /obj/item/weapon/gun/energy/staff))
```

Comment:
Sorry, this is on me, I meant *every energy gun* that can't be recharged should use the new system instead of typechecking. This is more consistent and maintainable. I should have been more clear, I thought it was obvious what I meant.

Reply:
Okay, I'll change it tomorrow